src/ucp/core/ucp_ep.c

```diff
@@ -1094,17 +1094,23 @@ void ucp_ep_set_failed(ucp_ep_h ucp_ep, ucp_lane_index_t lane,
                 close_req->send.flush.uct_flags |= UCT_FLUSH_FLAG_CANCEL;
                 ucp_ep_local_disconnect_progress(close_req);
             }
-        } else if (ep_ext_control->err_cb == NULL) {
+        } else if ((ep_ext_control->err_cb == NULL) ||
+                   (ucp_ep_config(ucp_ep)->key.err_mode ==
+                    UCP_ERR_HANDLING_MODE_NONE)) {
```

Comment:
should we make uct error callback (on the uct iface) return error to indicate the failure was not handled?

Reply:
yes, makes sense to me. pls, see the 2nd commit.
the 1st commit was updated to include fix for test failures (need to check only peer failure mode)